pkg/microservice/reaper/core/service/reaper/reaper.go

```diff
@@ -462,26 +462,15 @@ func (r *Reaper) AfterExec() error {
 			}
 			// if the given path is a directory
 			if info.IsDir() {
-				// we get ALL files in this directory and upload it to the object storage
-				files, err := ioutil.ReadDir(upload.FilePath)
+				err := client.UploadDir(r.Ctx.UploadStorageInfo.Bucket, upload.FilePath, upload.DestinationPath)
 				if err != nil {
-					return fmt.Errorf("failed to read file information in directory: [%s], the error is: %s", upload.FilePath, err)
-				}
-				for _, file := range files {
-					if !file.IsDir() {
-						key := filepath.Join(upload.DestinationPath, file.Name())
-						originalFilePath := filepath.Join(upload.FilePath, file.Name())
-						err := client.Upload(r.Ctx.UploadStorageInfo.Bucket, originalFilePath, key)
-						if err != nil {
-							fmt.Printf("Failed to upload [%s] to key [%s] on s3, the error is: %s", originalFilePath, key, err)
-						}
-					}
+					log.Infof("Failed to upload dir [%s] to path [%s] on s3, the error is: %s", upload.FilePath, upload.DestinationPath, err)
```

Comment:
`log.Infof()` -> `log.Errorf()` and if we encounter an error, we should return it.

Reply:
ok